python/cuml/feature_extraction/_vectorizers.py

```diff
@@ -602,15 +633,245 @@ def get_feature_names(self):
         """
         return self.vocabulary_
 
-    def _check_sklearn_params(self, analyzer, sklearn_params):
-        if callable(analyzer):
-            raise ValueError("cuML does not support callable analyzer,"
-                             " please refer to the cuML documentation for"
-                             " more information.")
 
-        for key, vals in sklearn_params.items():
-            if vals is not None:
-                raise TypeError("The Scikit-learn variable", key,
-                                " is not supported in cuML,"
-                                " please read the cuML documentation for"
-                                " more information.")
+class HashingVectorizer(_VectorizerMixin):
+    """Convert a collection of text documents to a matrix of token occurrences
+
+    It turns a collection of text documents into a cupy.sparse matrix holding
+    token occurrence counts (or binary occurrence information), possibly
+    normalized as token frequencies if norm='l1' or projected on the euclidean
+    unit sphere if norm='l2'.
+
+    This text vectorizer implementation uses the hashing trick to find the
+    token string name to feature integer index mapping.
+    This strategy has several advantages:
+    - it is very low memory scalable to large datasets as there is no need to
+      store a vocabulary dictionary in memory which even more important as GPU's are often memory constrained
+    - it is fast to pickle and un-pickle as it holds no state besides the
+      constructor parameters
+    - it can be used in a streaming (partial fit) or parallel pipeline as there
+      is no state computed during fit.
+    There are also a couple of cons (vs using a CountVectorizer with an
+    in-memory vocabulary):
+    - there is no way to compute the inverse transform (from feature indices to
+      string feature names) which can be a problem when trying to introspect
+      which features are most important to a model.
+    - there can be collisions: distinct tokens can be mapped to the same
+      feature index. However in practice this is rarely an issue if n_features
+      is large enough (e.g. 2 ** 18 for text classification problems).
+    - no IDF weighting as this would render the transformer stateful.
+    The hash function employed is the signed 32-bit version of Murmurhash3.
+    Parameters
```

Comment:
```suggestion
    
    Parameters
```

Reply:
Should be resolved by https://github.com/rapidsai/cuml/pull/2554/commits/145f59aff7996459080439df7395ac912f2b25ff